src/Ocelot/Cache/Middleware/OutputCacheMiddleware.cs

```diff
@@ -73,11 +74,18 @@ public async Task Invoke(HttpContext httpContext)
 
             var downstreamResponse = httpContext.Items.DownstreamResponse();
 
-            cached = await CreateCachedResponse(downstreamResponse);
+            if (downstreamResponse.StatusCode == HttpStatusCode.OK)
+            {
+                cached = await CreateCachedResponse(downstreamResponse);
 
-            _outputCache.Add(downStreamRequestCacheKey, cached, TimeSpan.FromSeconds(downstreamRoute.CacheOptions.TtlSeconds), downstreamRoute.CacheOptions.Region);
+                _outputCache.Add(downStreamRequestCacheKey, cached, TimeSpan.FromSeconds(downstreamRoute.CacheOptions.TtlSeconds), downstreamRoute.CacheOptions.Region);
 
-            Logger.LogDebug($"finished response added to cache for {downstreamUrlKey}");
+                Logger.LogDebug($"finished response added to cache for {downstreamUrlKey}");
+            }
+            else
+            {
+                Logger.LogDebug($"http request failed,  could not create cache for {downstreamUrlKey}");
```

Comment:
There's a double space. Also message could be better, something like "http request failed, skipped caching for {downstreamUrlKey}"